proposed/vector-instruction-set-default.md

```diff
@@ -0,0 +1,100 @@
+
+It is important to realize that this proposal is for a product that will be released in November 2021, and not likely used significantly until 2022. Early 2022 should be the primary time frame kept in mind. The Steam survey, for example, will report different numbers at that time. It's possible AVX2 will have reached today's AVX value by next year. To that point, PC sales have had a [surprising resurgence](https://www.idc.com/getdoc.jsp?containerId=prUS47274421) due to the pandemic, resulting in a lot of old machines being removed from the worldwide fleet (at least in countries with high GDP).
+
+An important degree of freedom is making different choices, per OS and hardware combination. Different OSes have a different hardware dynamic, different use cases, and a different future. Let's look at that. It's the heart of the proposal.
+
+## Differ behavior by OS
+
+It is reasonable to vary SIMD defaults, per operating system. There are many aspects of the product that differ by operating system or hardware platform. In fact, the JIT dynamically targeting different SIMD instructions per machine is such an example. For this case, the key consideration is the ready-to-run native code produced for a given operating system + CPU consideration. We could make one decision for Linux x64, another for Windows x64, and another for macOS Arm64. We already have to target different SIMD instructions for x64 vs Arm64, so targeting different SIMD instructions across OSes isn't much of a leap.
+
+It is useful to look at hardware dates again. Continuing with the Intel focus, we can see the following delivery of SIMD instructions in chips:
+
+* 2006: SSE4 was introduced with Core 2 x64 chips at the initial [Intel Core architecture](https://en.wikipedia.org/wiki/Intel_Core_(microarchitecture)) launch.
+* 2008: SSE4.2 was introduced with [Nehalem](https://en.wikipedia.org/wiki/Nehalem_(microarchitecture)) chips.
+* 2011: AVX was introduced with [Sandy Bridge](https://en.wikipedia.org/wiki/Sandy_Bridge) chips.
+* 2013: AVX2 was introduced with [Haswell](https://en.wikipedia.org/wiki/Haswell_(microarchitecture)) chips.
+
+Let's take a look at each operating system, assuming the Intel ISA.
+ 
+* On Windows, we need to be conservative. There are a lot of .NET users on Windows, and the majority of .NET desktop apps target Windows. Windows 10 requires AVX, but Win7 requires only SSE2. It is reasonable to expect that we'll continue supporting Windows 7 with .NET 6.0 (and probably not after that).
```

Comment:
Do we know when Win10 started requiring AVX (and if it is AVX or AVX2)?

The current docs still indicate it just requires SSE2, CMPXCHG16B, and a few other features: https://docs.microsoft.com/en-us/windows-hardware/design/minimum/minimum-hardware-requirements-overview#section-30---minimum-hardware-requirements-for-windows10-for-desktop-editions

AFAIK, Intel still produces some low power chips (such as the Atom lineup) which don't have AVX+ and which might appear in tablets or other "low" power devices.
(Certainly not the developer/server market, but it might be a factor for things like Paint.NET or other user-oriented .NET apps that might make their way to .NET 5+).

Reply:
I am likely wrong on the Windows AVX topic. I don't know where I read that.

The low-powered targets should be satisfied by targeting/using the 32-bit product or publishing (and recompiling) as self-contained.